src/main/java/ch/njol/skript/effects/EffDestroyableKeys.java

```diff
@@ -0,0 +1,85 @@
+package ch.njol.skript.effects;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import com.destroystokyo.paper.Namespaced;
+import org.bukkit.NamespacedKey;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class EffDestroyableKeys extends Effect {
+
+	static {
+		Skript.registerEffect(EffDestroyableKeys.class,
+			"allow %~itemtypes% to (destroy|break|mine) %itemtypes/namespacedkeys%",
+			"prevent %~itemtypes% from (destroy|break|mine)[ing] %itemtypes/namespacedkeys%");
+		// could these patterns be combined into 1 using a parse tag? like `[:dis]allow` or smth?
```

Comment:
sure, but i would keep the existing ones separate for clarity (and helps avoid `allow x from break y`)